torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -1202,6 +1249,78 @@ void PeepholeOptimizeShapeExpressions(Block* block) {
 
 } // anonymous namespace
 
+// This takes a _grad_sum_to_size output and tracks it to the return
+// statements that depend on it, checking that it only hits nodes
+// that commute with _grad_sum_to_size on its path.
+// If a vector pointer outputGradSumToSizes is passed, the sizes
+// will be recorded as target sizes for the outputs as applicable.
+// In the graph_fuser pass we only need to check that we can go to the
+// outputs while in the fuser's compiler we want to record the sizes.
+bool trackSingleGradSumToSizeToOutputs(
+    Value* gradSumToSizeOutput,
+    c10::optional<std::vector<int64_t>*> outputGradSumToSizes) {
```

Comment:
What's the point of taking an optional pointer? Can't you just use `nullptr` as the default value?